example/pool_provider.cpp

```diff
@@ -109,12 +109,12 @@ struct example {
 int main() {
   /*<<make injector with simple configuration>>*/
   // clang-format off
-  auto injector = di::make_injector<config>(
+  [[maybe_unused]] auto injector = di::make_injector<config>(
```

Comment:
will it work with C++14?

Reply:
It's a temporary wknd, still need some digging :disappointed: